src/hooks/useForm.ts

```diff
@@ -124,8 +133,27 @@ function useForm<T extends FormData>(
     setChanged(true);
   }
 
+  const setError = (field: keyof T, error: string | React.ReactNode) =>
+    setErrors(e => ({ ...e, [field]: error }));
+
+  const clearErrors = (field: keyof T | Array<keyof T>) => {
+    if (!field) {
+      setErrors({});
+    } else {
+      setErrors(e => {
+        (Array.isArray(field) ? field : [field]).forEach(name => {
+          delete e[name];
```

Comment:
`e` is a reference to the state object and you should never mutate it directly. Instead, copy/spread the object.